packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -354,31 +355,29 @@ function createMergedTransformStream(
   })
 }
 
+const CLOSE_TAG = '</body></html>'
+
 /**
  * This transform stream moves the suffix to the end of the stream, so results
  * like `</body></html><script>...</script>` will be transformed to
  * `<script>...</script></body></html>`.
  */
-function createMoveSuffixStream(
-  suffix: string
-): TransformStream<Uint8Array, Uint8Array> {
+function createMoveSuffixStream(): TransformStream<Uint8Array, Uint8Array> {
   let foundSuffix = false
 
-  const encodedSuffix = encoder.encode(suffix)
-
   return new TransformStream({
     transform(chunk, controller) {
       if (foundSuffix) {
         return controller.enqueue(chunk)
       }
 
-      const index = indexOfUint8Array(chunk, encodedSuffix)
+      const index = indexOfUint8Array(chunk, ENCODED_TAGS.CLOSED.BODY_AND_HTML)
       if (index > -1) {
         foundSuffix = true
 
         // If the whole chunk is the suffix, then don't write anything, it will
         // be written in the flush.
-        if (chunk.length === suffix.length) {
+        if (chunk.length === ENCODED_TAGS.CLOSED.BODY_AND_HTML.length) {
```

Comment:
This was a mistake. We should compare the lengths of two typed arrays, but `suffix` was the string data. Usually they should have the same length as the chars fit into ASCII (which is why we never had an error), but that will break when we have Unicode or change it to a different typed array.